src/contexts/W3iContext/hooks/authHooks.ts

```diff
@@ -7,8 +7,14 @@ import { formatEthChainsAddress, getChain, getEthChainAddress } from '../../../u
 export const useAuthState = (w3iProxy: Web3InboxProxy, proxyReady: boolean) => {
   const [accountQueryParam, setAccountQueryParam] = useState('')
 
-  const [userPubkey, setUserPubkey] = useState<string | undefined>(undefined)
-  const [authClient, setAuthClient] = useState<W3iAuthFacade | null>(null)
+  const [authClient, setAuthClient] = useState<W3iAuthFacade | null>(w3iProxy.auth)
+
+  const account = authClient?.getAccount()
+  const chain = authClient?.getChain()
+
+  const [userPubkey, setUserPubkey] = useState<string | undefined>(
+    formatEthChainsAddress(account, chain)
+  )
 
   useEffect(() => {
```

Comment:
This one and other effects on this hook, I'm not sure if they are really really necessary, need a bit more time to investigate.

Reply:
Lets handle this later since we fixed the wallet flash issue

Reply:
A lot of this is legacy carried over from when we had external providers, we can look into this later